### DISCUSSION TITLE
UR3 Support

### DISCUSSION THREAD

**Discussion by Yonahel** [visose/Robots#114] [Q&A] [answered]
Hello Robots contributors,

I'm using Robots to attempt to control a UR3 robotic arm. I noticed none of the provided 3dm and XML files include the UR3 robot arm, only the UR5 and UR10. I tried drafting my own XML, but ran into some troubles. I can't attach the files to this request, email me at ye2184@columbia.edu so I can forward you my XML and 3dm file. 

- For most of the parameters, it seems they added a tolerance of either 0.2, 0.3, or 0.7 mm. Is there any specific reason for this?
- I was able to map all the XML parameters to the CAD measurements for the 3 robots (UR3, UR5, UR10), but I could not find any dimension that matched revolute 4 on the UR5 or UR10 (I'm using the KULeuven XML for reference). Would you be able to help me find what this parameter is linked to?

I'm also missing splitting the UR3 on the 3dm file into indexed segments, but that's something I can work on by myself. Thanks for any help you can provide. 

Of course, if I get the file set up, I'd be happy to add my 3dm and XML file to the list of robots you have documented on this repo.

**Comment by Yonahel** [comment, marked as answer]
I can close this issue now, Vincent was able to reach out to me and help me with the file setup. Once I finish the 3dm file, I can submit my XML and 3dm file, if the project is interested in adding them.

**Reply by v01tt** [reply]
Hi Yonahel, 
Would you be able to share the files for UR3? I did not find it in the robot library.
Have you already included the 3dm and XML you've setup?

**Reply by Lorax-Hub** [reply]
I would also love to have those xml and 3dm files - our lab has two UR3s, and I would prefer to teach the undergrads on them rather than the ABB120s as they are safer. Since there are students in the class with both Macs and PCs, Robots seems like the best choice.

**Reply by Lorax-Hub** [reply]
Well, since I was in a time crunch, I threw together some files of my own - they're a little rough (based on .stl files for collision meshes from https://github.com/ros-industrial/universal_robot, so the link distances might be slightly off), but it should work in a pinch. I'm not sure if the values align correctly with the robot yet (checking that tomorrow), but these are the ones that seemed to work in the simulation...Would love any feedback/corrections if you have them!
[Lauras_UR3.zip](https://github.com/visose/Robots/files/9884998/Lauras_UR3.zip)
